### PR TITLE
NIFI-5916 Added an option to enable empty flowfiles to be sent if the…

### PR DESCRIPTION
**Pull request by MikeThomsen**
…re are no results from a query.

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java [hunk 1]
```diff
@@ -228,12 +249,19 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
...
+                    sent++;
                 }
             }
 
             if (input != null) {
                 session.transfer(input, REL_ORIGINAL);
             }
+
+            if (sent == 0 && sendEmpty) {
+                FlowFile empty = session.create(input);
```
Comment (mattyb149) [inline nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java]: Still tracking down whether there's a bug or intended behavior, but if `input == null` here it causes an NPE (looks like its due to changes in #2974 ). Just to be safe it should probably check for null here and just call `create()` instead. Would make a good unit test too (to show it works without an incoming flow file).

**Comment by mattyb149** [comment]
Lookin pretty good, left one comment, also please rebase while incorporating any changes, thanks!

File: nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java (hunk 1 above)
Comment (MikeThomsen) [reply nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java]: Ok. Fixed that.

**Comment by MikeThomsen** [comment]
@mattyb149 we're ready to merge.

**Comment by mattyb149** [comment]
+1 LGTM, ran unit and integration tests, and tested with a live NiFi instance, both with and without incoming flow files and with "output empty" set to both true and false. Thanks for the improvement! Merging to master

**Comment by Damien60** [comment]
Hello,
This behavior (NIFI-5916) was not extended to FetchGridFS. Is it possible to add it?
Thanks.

**Comment by mattyb149** [comment]
@Damien60 yes, please feel free to write up a Jira case to add this feature to FetchGridFS

**Pull request closed without merging**